promql/fuzz.go

```diff
@@ -0,0 +1,67 @@
+package promql
+// +build gofuzz
+
+/* PromQL parser fuzzing instrumentation for use with https://github.com/dvyukov/go-fuzz.
+ * 
+ * Fuzz each parser by building appropriately instrumented parser, ex. FuzzParseMetric and execute it with it's 
+ *
+ *     go-fuzz-build -func FuzzParseMetric -o FuzzParseMetric.zip github.com/prometheus/prometheus/promql
+ *
+ * And then run the tests with the appropriate inputs
+ *
+ *     go-fuzz -bin FuzzParseMetric.zip -workdir fuzz-data/ParseMetric
+ *
+ * Further input samples should go in the folders fuzz-data/ParseMetric/corpus.
+ *
+ * Repeat for ParseMetricSeletion, ParseExpr and ParseStmt
+ */
+
+const (
+	fuzz_interesting = 1
+	fuzz_meh         = 0
```

Comment:
In my (limited) experience, the return-value can have a large influence on how quickly the fuzzer converges on something interesting. I'd prefer including all, so the available options are more obvious (rather than a bunch of `return 0` and `return 1` scattered over the source code...)